.github/workflows/ci-ubuntu-24-sanitizer-nightly.yml

```diff
@@ -11,15 +11,18 @@ on:
 
 # Cancel any preceding run on the pull request.
 concurrency:
-  group: test-ubuntu24-clang-sanitizers-${{ github.event.pull_request.number || github.ref }}
+  group: test-ubuntu-clang-sanitizers-${{ github.event.pull_request.number || github.ref }}
   cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
 
 jobs:
-  # Build with clang and test p4c on Ubuntu 24.04.
-  test-ubuntu24-clang-sanitizers:
+  # Build with clang and test p4c on Ubuntu
+  test-ubuntu-clang-sanitizers:
     # Only run on pull requests with the "run-sanitizer" label.
     if: ${{ github.event_name == 'schedule' || contains(github.event.pull_request.labels.*.name, 'run-sanitizer') }}
-    runs-on: ubuntu-24.04
+    strategy:
+      matrix:
+        os: [ubuntu-22.04, ubuntu-24.04]
```

Comment:
Given this generalization to other versions, maybe best if the name of the file changes, too?